spec/std/spec/cli_spec.cr

```diff
@@ -0,0 +1,183 @@
+
+    def testing
+      @testing = true
+      yield
+    ensure
+      @testing = false
+    end
+
+    private def display(message)
+      stdout.puts message
+
+      if testing?
+        @exited = 0
+      else
+        exit
+      end
+    end
+
+    private def terminate(message, status = 1)
+      stderr.puts message
+
+      if testing?
+        @exited = status
+      else
+        exit status
+      end
+    end
+  end
+end
+
+describe Spec::CLI do
+  context "parses options" do
+    it "captures provided example pattern" do
+      _, options = prepare_cli %w(-e Foo)
+      options.pattern?.should be_truthy
+      options.pattern.should eq("Foo")
+
+      _, options = prepare_cli %w(--example Bar)
+      options.pattern?.should be_truthy
+      options.pattern.should eq("Bar")
+    end
+
+    it "captures provided line number" do
+      _, options = prepare_cli %w(-l 10)
+      options.line?.should be_truthy
+      options.line.should eq(10)
+
+      _, options = prepare_cli %w(--line 10)
+      options.line?.should be_truthy
+      options.line.should eq(10)
+    end
+
+    it "enables slowest mode" do
+      _, options = prepare_cli %w(-p)
+      options.slowest?.should be_truthy
+      options.slowest.should eq(10)
+
+      _, options = prepare_cli %w(--profile)
+      options.slowest?.should be_truthy
+      options.slowest.should eq(10)
+    end
+
+    it "enables fail-fast mode" do
+      _, options = prepare_cli %w(--fail-fast)
+      options.fail_fast?.should be_true
+    end
+
+    context "locations" do
+      it "captures single file:line location" do
+        _, options = prepare_cli %w(--location foo.cr:1)
+
+        options.locations.size.should eq(1)
+        options.locations.should eq([{"foo.cr", 1}])
+      end
+
+      it "captures multiple file:line locations" do
+        _, options = prepare_cli %w(--location foo.cr:10 --location bar.cr:50)
+
+        options.locations.size.should eq(2)
+        options.locations.should eq([{"foo.cr", 10}, {"bar.cr", 50}])
+      end
+
+      it "aborts on incorrect location format" do
+        io = IO::Memory.new
+
+        cli = Spec::CLI.new %w(--location missing.cr), stderr: io
+        cli.testing do
+          cli.prepare
+        end
+
+        cli.options.locations.size.should eq(0)
+        cli.exited?.should be_truthy
+        cli.exited.should eq(1)
+
+        io.rewind
+        io.gets_to_end.should contain("location missing.cr must be file:line")
+      end
+    end
+
+    it "adds JUnit format to output" do
+      _, options = prepare_cli %w(--junit_output tmp)
+
+      options.formatters.size.should eq(1)
+      options.formatters.first.should be_a(Spec::JUnitFormatter)
+    end
+
+    it "displays help options" do
+      io = IO::Memory.new
+
+      cli = Spec::CLI.new %w(--help), stdout: io
+      cli.testing do
+        cli.prepare
+      end
+
+      cli.exited?.should be_truthy
+      cli.exited.should eq(0)
+
+      io.rewind
+      output = io.gets_to_end
```

Comment:
Instead of:
```cr
io.rewind
output = io.gets_to_end
```
You can do: `output = io.to_s`

(and ditto everywhere you use this)

Reply:
Completely forgot about that, will update. Thank you.